homeassistant/components/eheimdigital/fan.py

```diff
@@ -0,0 +1,135 @@
+"""EHEIM Digital fans."""
+
+from typing import Any, override
+
+from eheimdigital.classic_vario import EheimDigitalClassicVario
+from eheimdigital.device import EheimDigitalDevice
+from eheimdigital.types import EheimDigitalClientError, FilterMode
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import EheimDigitalConfigEntry
+from .const import FILTER_BIO_MODE, FILTER_PRESET_TO_FILTER_MODE, FILTER_PULSE_MODE
+from .coordinator import EheimDigitalUpdateCoordinator
+from .entity import EheimDigitalEntity
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: EheimDigitalConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the callbacks for the coordinator so fans can be added as devices are found."""
+
+    coordinator = entry.runtime_data
+
+    def async_setup_device_entities(
+        device_address: str | dict[str, EheimDigitalDevice],
+    ) -> None:
+        """Set up the climate entities for one or multiple devices."""
+        entities: list[EheimDigitalClassicVarioPump] = []
+        if isinstance(device_address, str):
+            device_address = {device_address: coordinator.hub.devices[device_address]}
```

Comment:
Couldn't this be moved in the coordinator, so it always provides the `dict` as needed? This bit of code is duplicated across all the platforms and I didn't see any instance where it's used as a plain string.